network/v1alpha1/types_dnsnameresolver.go

```diff
@@ -0,0 +1,119 @@
+type DNSNameResolverSpec struct {
+	// name is the DNS name for which the DNS name resolution information will be stored.
+	// For a regular DNS name, only the DNS name resolution information of the regular DNS
+	// name will be stored. For a wildcard DNS name, the DNS name resolution information
+	// of all the DNS names, that matches the wildcard DNS name, will be stored.
+	// For a wildcard DNS name, the '*' will match only one label. Additionally, only a single
+	// '*' can be used at the beginning of the wildcard DNS name. For example, '*.example.com.'
+	// will match 'sub1.example.com.' but won't match 'sub2.sub1.example.com.'
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=^(\*\.)?([A-Za-z0-9-]+\.)*[A-Za-z0-9-]+\.$
+	// +kubebuilder:validation:MaxLength=254
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="spec.name is immutable"
+	Name string `json:"name"`
+}
+
+// DNSNameResolverStatus defines the observed status of DNSNameResolver.
+type DNSNameResolverStatus struct {
+	// resolvedNames contains a list of matching DNS names and their corresponding IP addresses
+	// along with TTL and last DNS lookup time.
+	// +listType=map
+	// +listMapKey=dnsName
+	// +patchMergeKey=dnsName
+	// +patchStrategy=merge
+	// +optional
+	ResolvedNames []DNSNameResolverStatusItem `json:"resolvedNames,omitempty" patchStrategy:"merge" patchMergeKey:"dnsName"`
+}
+
+// +kubebuilder:validation:Pattern=`^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\/([0-9]|[1-2][0-9]|3[0-2])$|^s*((([0-9A-Fa-f]{1,4}:){7}(:|([0-9A-Fa-f]{1,4})))|(([0-9A-Fa-f]{1,4}:){6}:([0-9A-Fa-f]{1,4})?)|(([0-9A-Fa-f]{1,4}:){5}(((:[0-9A-Fa-f]{1,4}){0,1}):([0-9A-Fa-f]{1,4})?))|(([0-9A-Fa-f]{1,4}:){4}(((:[0-9A-Fa-f]{1,4}){0,2}):([0-9A-Fa-f]{1,4})?))|(([0-9A-Fa-f]{1,4}:){3}(((:[0-9A-Fa-f]{1,4}){0,3}):([0-9A-Fa-f]{1,4})?))|(([0-9A-Fa-f]{1,4}:){2}(((:[0-9A-Fa-f]{1,4}){0,4}):([0-9A-Fa-f]{1,4})?))|(([0-9A-Fa-f]{1,4}:){1}(((:[0-9A-Fa-f]{1,4}){0,5}):([0-9A-Fa-f]{1,4})?))|(:(:|((:[0-9A-Fa-f]{1,4}){1,7}))))(%.+)?s*/([0-9]|[1-9][0-9]|1[0-1][0-9]|12[0-8])$`
+// IPAddressStr is used for validation of an IP address.
+type IPAddressStr string
+
+// DNSNameResolverStatusItem describes the details of a resolved DNS name.
+type DNSNameResolverStatusItem struct {
+	// conditions provide information about the state of the DNS name.
+	// Known .status.conditions.type is: "Degraded"
+	// +optional
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+	// dnsName is the resolved DNS name matching the name field of DNSNameResolverSpec.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=^(\*\.)?([A-Za-z0-9-]+\.)*[A-Za-z0-9-]+\.$
+	// +kubebuilder:validation:MaxLength=254
+	DNSName string `json:"dnsName"`
+	// info gives the list of associated IP addresses and the corresponding TTL and last
+	// lookup time for the dnsName.
```

Comment:
Want to make it clear the last lookup and ttl belong to each IP address
```suggestion
	// info gives the list of associated IP addresses and their corresponding TTLs and last
	// lookup times for the dnsName.
```